### PR TITLE
Dockerfile example simplification

### PR DESCRIPTION
**Pull request by tbavelier**
### What does this PR do?

* Updates the `Dockerfile` to use the easier installation steps in which `agent integration` command can directly pull the wheel without having to `git clone`
    * There is still a "manual" step to download the Python dependencies and install them with the Agent embedded PIP
* Updates `README.md` to reflect the new `Dockerfile` example, which does not include build arguments

### Motivation

* Simplify setup for customers

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: nvml/tests/Dockerfile [hunk 1]
```diff
@@ -1,39 +1,14 @@
...
-ARG INTEGRATIONS_VERSION=master
+# As the NVML integration is a community integration, it is not packaged with the Agent image by default. A custom image with the packaged integration should therefore be used in containerized environments.
+# Setup taken from https://docs.datadoghq.com/agent/guide/use-community-integrations/?tab=docker
 
-FROM python:${PY_VERSION} AS wheel_builder
-ARG INTEGRATIONS_VERSION
-ARG DD_AGENT_VERSION
-WORKDIR /wheels
+# Installs the NVML integration version 1.0.6.
+# Other available versions are available by reviewing the changelog (https://github.com/DataDog/integrations-extras/blob/master/nvml/CHANGELOG.md)
```
Comment (maycmlee) [inline nvml/tests/Dockerfile:5]:
```suggestion
# See the changelog for the other versions that are available (https://github.com/DataDog/integrations-extras/blob/master/nvml/CHANGELOG.md)
```

File: nvml/tests/Dockerfile [hunk 2]
```diff
@@ -1,39 +1,14 @@
...
-
-# Work around bug https://github.com/DataDog/datadog-agent/issues/4142
-RUN ln -s /etc/datadog-agent/datadog-docker.yaml /etc/datadog-agent/datadog.yaml && \
-	agent integration install -r -w /dist/*.whl && \
-	rm /etc/datadog-agent/datadog.yaml
-
-# The command 'agent integration install' does not capture dependencies.  This is strange
-# to me.  Install those on their own
-RUN /opt/datadog-agent/embedded/bin/pip3 install -r /requirements.in
+# Install PIP libraries used by the check using a permalink to the requirements
```
Comment (maycmlee) [inline nvml/tests/Dockerfile:9]:
```suggestion
# Use the permalink to the requirements to install the PIP libraries used by the check
```

**Review by maycmlee — COMMENTED** [review]
Left a couple of suggestions!

**Review by maycmlee — APPROVED** [review]

**Comment by therc** [comment]
I was trying to troubleshoot and fix a regression with nvml and agent 7.47, when I noticed that my custom agent images were not picking up my nvml changes. Eventually, I found this PR. Who is it actually serving? Because it removed INTEGRATIONS_VERSION, I can no longer test a custom, unmerged branch. How is one supposed to develop and test the extension? (I also had a local patch to use a fork, rather than DataDog/integrations-extras).

Can this change be rolled back? Or should I create a separate development Dockerfile?

**Comment by tbavelier** [comment]
@therc Please feel free to make a new PR to modify the Dockerfile appropriately keeping in mind the idea was to rely on `agent integration install datadog-nvml ...` to facilitate the onboarding of this integration. But considering you'd have un-released, un-merged changes, you'd indeed need to have a separate development Dockerfile to copy over your changes for testing purposes in any case.

One additional point I would like to mention is that Datadog also released in the Core repo an Nvidia DCGM Exporter integration (https://github.com/DataDog/integrations-core/tree/master/dcgm) which is officially supported by Datadog (Core repo) and relies on Nvidia guidance (use of their DCGM exporter : https://github.com/NVIDIA/dcgm-exporter), which should have parity I believe in metrics collected.

**Pull request merged**